hazelcast/src/main/java/com/hazelcast/cardinality/CardinalityEstimator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cardinality;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.spi.annotation.Beta;
+
+/**
+ * CardinalityEstimator is a redundant and highly available distributed data-structure used
+ * for probabilistic cardinality estimation purposes, on unique items, in significantly sized data cultures.
+ *
+ * CardinalityEstimator is internally based on a HyperLogLog++ data-structure,
+ * and uses P^2 byte registers for storage and computation. (Default P = 14)
+ */
+@Beta
+public interface CardinalityEstimator extends DistributedObject {
+
+    /**
+     * Consider the 64bit hash value, produced by the serialized form of the given object,
+     * in the cardinality estimation. The implementation is free to choose whether this hash
+     * will be used towards the estimation, therefore, this operation may or may not affect
+     * the current estimation.
+     *
+     * Input object, unless a boxed primitive, must have a
+     * {@link com.hazelcast.nio.serialization.StreamSerializer} implementation, registered in the Hazelcast
```

Comment:
This is confusing/misleading. 
The object also can just implement `j.i.Serializable`, it can implement `DataSerializable`, etc. The only condition is this: "Hazelcast has to know how to serialize it". The exact serialization strategy is irrelevant. I would completely remove this as this is true for (almost) all other Hazelcast calls. Even the most basic - `map.put()`, etc.